jobs openrewrite-static-analysis is not working

https://app.circleci.com/pipelines/github/checkstyle/checkstyle/48161/workflows/f09076d7-4b4f-46fa-9755-3e89f614dd21/jobs/1648035

```
[INFO] --- rewrite:6.40.0:run (default-cli) @ checkstyle ---
[INFO] Using active recipe(s) [org.checkstyle.StaticAnalysis]
[INFO] Using active styles(s) []
[INFO] Validating active recipes...
[ERROR] Recipe validation error in org.checkstyle.StaticAnalysis for property org.openrewrite.staticanalysis.CodeCleanup: Recipe class org.openrewrite.staticanalysis.CodeCleanup cannot be found
[ERROR] Recipe validation error in org.checkstyle.StaticAnalysis for property org.openrewrite.staticanalysis.CommonStaticAnalysis: Recipe class org.openrewrite.staticanalysis.CommonStaticAnalysis cannot be found
[ERROR] Recipe validation errors detected as part of one or more activeRecipe(s). Execution will continue regardless.
[INFO] Skipping Maven parsing...
[INFO] Project [checkstyle] Parsing source files
[WARNING] There were problems parsing some source files
[WARNING] There were problems parsing .ci/checker-framework.groovy
[WARNING] org.openrewrite.groovy.GroovyParsingException: Failed to parse .ci/checker-framework.groovy at cursor position 14601. The surrounding characters in the original source are:
```

--- a/config/rewrite.yml
+++ b/config/rewrite.yml
@@ -58,7 +58,6 @@ recipeList:
   # composite list org.openrewrite.staticanalysis.CodeCleanup
   # https://docs.openrewrite.org/recipes/staticanalysis/codecleanup
   # ###
-  # all suppressions until https://github.com/checkstyle/checkstyle/issues/18952
   # Kept excluded: OpenRewrite's DefaultComesLast always moves default to
   # the end of a switch statement, ignoring Checkstyle's
   # DefaultComesLastCheck option `skipIfLastAndSharedWithCase`. When set
@@ -76,8 +75,7 @@ recipeList:
   # loop update clauses, but Checkstyle has no equivalent check for this.
   # Both forms are functionally identical in for loops, and some style
   # guides prefer pre-increment.
-  # until https://github.com/checkstyle/checkstyle/issues/18952
-  #- org.openrewrite.staticanalysis.ForLoopControlVariablePostfixOperators
+  - org.openrewrite.staticanalysis.ForLoopControlVariablePostfixOperators
   # Kept excluded: OpenRewrite's FinalizePrivateFields automatically adds
   # final to private instance fields, but Checkstyle has no equivalent
   # check for private fields (FinalLocalVariable only covers local

--- a/config/rewrite.yml
+++ b/config/rewrite.yml
@@ -54,75 +54,173 @@ type: specs.openrewrite.org/v1beta/recipe
 name: org.checkstyle.StaticAnalysis
 description: OpenRewrite standard fixes.
 recipeList:
-  # composite list
-  - org.openrewrite.staticanalysis.CodeCleanup:
-      exclude:
-        # all suppressions until https://github.com/checkstyle/checkstyle/issues/18952
-        # Kept excluded: OpenRewrite's DefaultComesLast always moves default to
-        # the end of a switch statement, ignoring Checkstyle's
-        # DefaultComesLastCheck option `skipIfLastAndSharedWithCase`. When set
-        # to true, Checkstyle permits default to share a case group
-        # (e.g. case 1: default: break;), which OpenRewrite would incorrectly
-        # reorder.
-        - org.openrewrite.staticanalysis.DefaultComesLast
-        # Kept excluded: OpenRewrite's EmptyBlock removes blocks entirely,
-        # which conflicts with Checkstyle's EmptyBlockCheck that requires
-        # blocks to have statements or text. Automatic deletion is too aggressive.
-        - org.openrewrite.staticanalysis.EmptyBlock
-        # Kept excluded: OpenRewrite's FinalizePrivateFields automatically adds
-        # final to private instance fields, but Checkstyle has no equivalent
-        # check for private fields (FinalLocalVariable only covers local
-        # variables). Adding final to fields can break frameworks that mutate
-        # fields via reflection such as Spring, Hibernate, and Mockito.
-        - org.openrewrite.staticanalysis.FinalizePrivateFields
-        # Kept excluded: OpenRewrite's ForLoopControlVariablePostfixOperators
-        # always replaces pre-increment (++i) with post-increment (i++) in for
-        # loop update clauses, but Checkstyle has no equivalent check for this.
-        # Both forms are functionally identical in for loops, and some style
-        # guides prefer pre-increment.
-        - org.openrewrite.staticanalysis.ForLoopControlVariablePostfixOperators
-        # conflicts with MethodParamPadCheck in checkstyle_checks.xml
-        - org.openrewrite.java.format.MethodParamPad
-        # conflicts with NoWhitespaceAfterCheck in checkstyle_checks.xml
-        - org.openrewrite.java.format.NoWhitespaceAfter
-        # conflicts with NoWhitespaceBeforeCheck in checkstyle_checks.xml
-        - org.openrewrite.java.format.NoWhitespaceBefore
-        # Kept excluded: OpenRewrite's TypecastParenPad always removes spaces around
-        # typecast parentheses, while Checkstyle's TypecastParenPadCheck allows
-        # configuration via the `option` property (`space` or `nospace`).
-        - org.openrewrite.staticanalysis.TypecastParenPad
-        # Kept excluded: OpenRewrite's ExplicitInitialization automatically removes
-        # explicit initializations to default values (null, 0, false), while
-        # Checkstyle's ExplicitInitializationCheck only reports violations,
-        # allowing developers to decide whether to keep them for clarity.
-        # Checkstyle also provides the `onlyObjectReferences` option to limit
-        # checks to object references only.
-        - org.openrewrite.staticanalysis.ExplicitInitialization
-        # Kept excluded: OpenRewrite's FallThrough always adds break statements,
-        # ignoring Checkstyle's FallThroughCheck which allows configurable relief
-        # comments via `reliefPattern` and can optionally skip checking the last
-        # case group with `checkLastCaseGroup`
-        - org.openrewrite.staticanalysis.FallThrough
-        - org.openrewrite.staticanalysis.HideUtilityClassConstructor
-        # Kept excluded: OpenRewrite's NeedBraces always adds braces, overriding
-        # Checkstyle's NeedBracesCheck which allows configuration for single-line
-        # statements and empty loop bodies via `allowSingleLineStatement` and
-        # `allowEmptyLoopBody` options.
-        - org.openrewrite.staticanalysis.NeedBraces
-        # Kept excluded: OpenRewrite's OperatorWrap forces operators to the end of
-        # the line with no configuration, while Checkstyle's OperatorWrapCheck
-        # allows projects to choose between `nl` (new line) and `eol` (end of line)
-        # via the `option` property, and can limit checks to specific tokens.
-        - org.openrewrite.staticanalysis.OperatorWrap
-        - org.openrewrite.staticanalysis.ReplaceThreadRunWithThreadStart
-        - org.openrewrite.staticanalysis.ChainStringBuilderAppendCalls
-        # Kept excluded: to avoid conflicts with CustomImportOrderCheck
-        - org.openrewrite.staticanalysis.CustomImportOrder
-  - org.openrewrite.staticanalysis.CommonStaticAnalysis:
-      exclude:
-        # there is some problem in their implementation, let us revisit it in few years.
-        # https://github.com/checkstyle/checkstyle/pull/18903#discussion_r3215832346
-        - org.openrewrite.staticanalysis.LambdaBlockToExpression
+  # ###
+  # composite list org.openrewrite.staticanalysis.CodeCleanup
+  # https://docs.openrewrite.org/recipes/staticanalysis/codecleanup
+  # ###
+  # all suppressions until https://github.com/checkstyle/checkstyle/issues/18952
+  # Kept excluded: OpenRewrite's DefaultComesLast always moves default to
+  # the end of a switch statement, ignoring Checkstyle's
+  # DefaultComesLastCheck option `skipIfLastAndSharedWithCase`. When set
+  # to true, Checkstyle permits default to share a case group
+  # (e.g. case 1: default: break;), which OpenRewrite would incorrectly
+  # reorder.
+  # - org.openrewrite.staticanalysis.DefaultComesLast
+  # Kept excluded: OpenRewrite's EmptyBlock removes blocks entirely,
+  # which conflicts with Checkstyle's EmptyBlockCheck that requires
+  # blocks to have statements or text. Automatic deletion is too aggressive.
+  # - org.openrewrite.staticanalysis.EmptyBlock
+  - org.openrewrite.java.format.EmptyNewlineAtEndOfFile
+  # Kept excluded: OpenRewrite's ForLoopControlVariablePostfixOperators
+  # always replaces pre-increment (++i) with post-increment (i++) in for
+  # loop update clauses, but Checkstyle has no equivalent check for this.
+  # Both forms are functionally identical in for loops, and some style
+  # guides prefer pre-increment.
+  # until https://github.com/checkstyle/checkstyle/issues/18952
+  #- org.openrewrite.staticanalysis.ForLoopControlVariablePostfixOperators
+  # Kept excluded: OpenRewrite's FinalizePrivateFields automatically adds
+  # final to private instance fields, but Checkstyle has no equivalent
+  # check for private fields (FinalLocalVariable only covers local
+  # variables). Adding final to fields can break frameworks that mutate
+  # fields via reflection such as Spring, Hibernate, and Mockito.
+  # - org.openrewrite.staticanalysis.FinalizePrivateFields
+  # conflicts with MethodParamPadCheck in checkstyle_checks.xml
+  # - org.openrewrite.java.format.MethodParamPad
+  # conflicts with NoWhitespaceAfterCheck in checkstyle_checks.xml
+  # - org.openrewrite.java.format.NoWhitespaceAfter
+  # conflicts with NoWhitespaceBeforeCheck in checkstyle_checks.xml
+  # - org.openrewrite.java.format.NoWhitespaceBefore
+  - org.openrewrite.java.format.PadEmptyForLoopComponents
+  # Kept excluded: OpenRewrite's TypecastParenPad always removes spaces around
+  # typecast parentheses, while Checkstyle's TypecastParenPadCheck allows
+  # configuration via the `option` property (`space` or `nospace`).
+  # - org.openrewrite.staticanalysis.TypecastParenPad
+  - org.openrewrite.staticanalysis.EqualsAvoidsNull
+  # Kept excluded: OpenRewrite's ExplicitInitialization automatically removes
+  # explicit initializations to default values (null, 0, false), while
+  # Checkstyle's ExplicitInitializationCheck only reports violations,
+  # allowing developers to decide whether to keep them for clarity.
+  # Checkstyle also provides the `onlyObjectReferences` option to limit
+  # checks to object references only.
+  # - org.openrewrite.staticanalysis.ExplicitInitialization
+  # Kept excluded: OpenRewrite's FallThrough always adds break statements,
+  # ignoring Checkstyle's FallThroughCheck which allows configurable relief
+  # comments via `reliefPattern` and can optionally skip checking the last
+  # case group with `checkLastCaseGroup`
+  # - org.openrewrite.staticanalysis.FallThrough
+  - org.openrewrite.staticanalysis.HideUtilityClassConstructor
+  # Kept excluded: OpenRewrite's NeedBraces always adds braces, overriding
+  # Checkstyle's NeedBracesCheck which allows configuration for single-line
+  # statements and empty loop bodies via `allowSingleLineStatement` and
+  # `allowEmptyLoopBody` options.
+  # - org.openrewrite.staticanalysis.NeedBraces
+  # Kept excluded: OpenRewrite's OperatorWrap forces operators to the end of
+  # the line with no configuration, while Checkstyle's OperatorWrapCheck
+  # allows projects to choose between `nl` (new line) and `eol` (end of line)
+  # via the `option` property, and can limit checks to specific tokens.
+  # - org.openrewrite.staticanalysis.OperatorWrap
+  - org.openrewrite.staticanalysis.UnnecessaryParentheses
+  - org.openrewrite.staticanalysis.ReplaceThreadRunWithThreadStart
+  - org.openrewrite.staticanalysis.ChainStringBuilderAppendCalls
+  - org.openrewrite.staticanalysis.RemoveMethodsOnlyCallSuper
+  - org.openrewrite.staticanalysis.ReplaceStringBuilderWithString
+  - org.openrewrite.java.ShortenFullyQualifiedTypeReferences
+  - org.openrewrite.java.SimplifySingleElementAnnotation
+  # conflict with ImportOrder Check
+  # - org.openrewrite.java.OrderImports
+  - org.openrewrite.staticanalysis.RemoveUnusedLabels
+
+  # ###
+  # composite list org.openrewrite.staticanalysis.CommonStaticAnalysis
+  # https://docs.openrewrite.org/recipes/staticanalysis/commonstaticanalysis
+  # ###
+  - org.openrewrite.staticanalysis.AbstractClassPublicConstructor
+  - org.openrewrite.staticanalysis.AtomicPrimitiveEqualsUsesGet
+  - org.openrewrite.staticanalysis.BigDecimalDoubleConstructorRecipe
+  - org.openrewrite.staticanalysis.BigDecimalRoundingConstantsToEnums
+  - org.openrewrite.staticanalysis.BooleanChecksNotInverted
+  - org.openrewrite.staticanalysis.CaseInsensitiveComparisonsDoNotChangeCase
+  - org.openrewrite.staticanalysis.CatchClauseOnlyRethrows
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.ChainStringBuilderAppendCalls
+  - org.openrewrite.staticanalysis.CollectionToArrayShouldHaveProperType
+  - org.openrewrite.staticanalysis.CovariantEquals
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.DefaultComesLast
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.EmptyBlock
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.EqualsAvoidsNull
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.ExplicitInitialization
+  - org.openrewrite.staticanalysis.ExternalizableHasNoArgsConstructor
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.FinalizePrivateFields
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.FallThrough
+  - org.openrewrite.staticanalysis.FinalClass
+  - org.openrewrite.staticanalysis.FixStringFormatExpressions
+  - org.openrewrite.staticanalysis.ForLoopIncrementInUpdate
+  - org.openrewrite.staticanalysis.IndexOfChecksShouldUseAStartPosition
+  - org.openrewrite.staticanalysis.IndexOfReplaceableByContains
+  - org.openrewrite.staticanalysis.IndexOfShouldNotCompareGreaterThanZero
+  - org.openrewrite.staticanalysis.InlineVariable
+  - org.openrewrite.staticanalysis.IsEmptyCallOnCollections
+  # there is some problem in their implementation, let us revisit it in few years.
+  # https://github.com/checkstyle/checkstyle/pull/18903#discussion_r3215832346
+  # - org.openrewrite.staticanalysis.LambdaBlockToExpression
+  - org.openrewrite.staticanalysis.MethodNameCasing
+  - org.openrewrite.staticanalysis.MinimumSwitchCases
+  - org.openrewrite.staticanalysis.ModifierOrder
+  - org.openrewrite.staticanalysis.MultipleVariableDeclarations
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.NeedBraces
+  - org.openrewrite.staticanalysis.NestedEnumsAreNotStatic
+  - org.openrewrite.staticanalysis.NewStringBuilderBufferWithCharArgument
+  - org.openrewrite.staticanalysis.NoDoubleBraceInitialization
+  - org.openrewrite.staticanalysis.NoEmptyCollectionWithRawType
+  - org.openrewrite.staticanalysis.NoEqualityInForCondition
+  - org.openrewrite.staticanalysis.NoFinalizer
+  - org.openrewrite.staticanalysis.NoPrimitiveWrappersForToStringOrCompareTo
+  - org.openrewrite.staticanalysis.NoRedundantJumpStatements
+  - org.openrewrite.staticanalysis.NoToStringOnStringType
+  - org.openrewrite.staticanalysis.NoValueOfOnStringType
+  - org.openrewrite.staticanalysis.ObjectFinalizeCallsSuper
+  - org.openrewrite.staticanalysis.PreferSystemGetPropertyOverGetenv
+  - org.openrewrite.staticanalysis.PrimitiveWrapperClassConstructorToValueOf
+  - org.openrewrite.staticanalysis.RedundantFileCreation
+  - org.openrewrite.staticanalysis.RemoveExtraSemicolons
+  - org.openrewrite.staticanalysis.RemoveRedundantNullCheckBeforeInstanceof
+  - org.openrewrite.staticanalysis.RemoveRedundantNullCheckBeforeLiteralEquals
+  - org.openrewrite.staticanalysis.RenameMethodsNamedHashcodeEqualOrToString
+  - org.openrewrite.staticanalysis.ReplaceClassIsInstanceWithInstanceof
+  - org.openrewrite.staticanalysis.ReplaceLambdaWithMethodReference
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.ReplaceStringBuilderWithString
+  - org.openrewrite.staticanalysis.ReplaceStringConcatenationWithStringValueOf
+  - org.openrewrite.staticanalysis.SimplifyArraysAsList
+  - org.openrewrite.staticanalysis.SimplifyBooleanExpression
+  - org.openrewrite.staticanalysis.SimplifyBooleanReturn
+  - org.openrewrite.staticanalysis.StaticMethodNotFinal
+  - org.openrewrite.staticanalysis.StringLiteralEquality
+  - org.openrewrite.staticanalysis.UnnecessaryCloseInTryWithResources
+  - org.openrewrite.staticanalysis.UnnecessaryExplicitTypeArguments
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.staticanalysis.UnnecessaryParentheses
+  - org.openrewrite.staticanalysis.UnnecessaryPrimitiveAnnotations
+  - org.openrewrite.staticanalysis.UnnecessaryReturnAsLastStatement
+  - org.openrewrite.staticanalysis.UpperCaseLiteralSuffixes
+  - org.openrewrite.staticanalysis.UseDiamondOperator
+  - org.openrewrite.staticanalysis.UseJavaStyleArrayDeclarations
+  - org.openrewrite.staticanalysis.UsePortableNewlines
+  - org.openrewrite.staticanalysis.WhileInsteadOfFor
+  - org.openrewrite.staticanalysis.WriteOctalValuesAsDecimal
+  - org.openrewrite.kotlin.cleanup.EqualsMethodUsage
+  - org.openrewrite.kotlin.cleanup.ImplicitParameterInLambda
+  - org.openrewrite.kotlin.cleanup.ReplaceCharToIntWithCode
+  # managed by org.openrewrite.staticanalysis.CodeCleanup
+  # - org.openrewrite.java.OrderImports
+
   # individual recipes
   - org.openrewrite.java.RemoveUnusedImports
   - org.openrewrite.java.format.NormalizeFormat

--- a/pom.xml
+++ b/pom.xml
@@ -666,6 +666,7 @@
             <configLocation>config/rewrite.yml</configLocation>
             <exportDatatables>true</exportDatatables>
             <skipMavenParsing>true</skipMavenParsing>
+            <failOnInvalidActiveRecipes>true</failOnInvalidActiveRecipes>
             <activeRecipes>
               <recipe>org.checkstyle.AllAutoFixes</recipe>
             </activeRecipes>

--- a/src/it/java/com/google/checkstyle/test/base/AbstractIndentationTestSupport.java
+++ b/src/it/java/com/google/checkstyle/test/base/AbstractIndentationTestSupport.java
@@ -261,7 +261,7 @@ public abstract class AbstractIndentationTestSupport extends AbstractGoogleModul
      */
     private static int getLineStart(String line, final int tabWidth) {
         int lineStart = 0;
-        for (int index = 0; index < line.length(); ++index) {
+        for (int index = 0; index < line.length(); index++) {
             if (!Character.isWhitespace(line.charAt(index))) {
                 lineStart = CommonUtil.lengthExpandedTabs(line, index, tabWidth);
                 break;

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java
@@ -398,7 +398,7 @@ public class SuppressWarningsHolder
      */
     private static DetailAST getNthChild(DetailAST ast, int index) {
         DetailAST child = ast.getFirstChild();
-        for (int i = 0; i < index && child != null; ++i) {
+        for (int i = 0; i < index && child != null; i++) {
             child = child.getNextSibling();
         }
         return child;

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocContentLocationCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocContentLocationCheck.java
@@ -186,7 +186,7 @@ public class JavadocContentLocationCheck extends AbstractCheck {
     private static int findIndexOfFirstNonBlankLine(String commentContent) {
         int lineNo = 0;
         boolean noContent = true;
-        for (int i = 0; i < commentContent.length(); ++i) {
+        for (int i = 0; i < commentContent.length(); i++) {
             final char character = commentContent.charAt(i);
             if (character == '\n') {
                 ++lineNo;

--- a/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java
@@ -490,7 +490,7 @@ public final class CheckUtil {
                                                        String classToBeMatched) {
         final int length = Math.min(classToBeMatched.length(), patternClass.length());
         int result = 0;
-        for (int i = 0; i < length && patternClass.charAt(i) == classToBeMatched.charAt(i); ++i) {
+        for (int i = 0; i < length && patternClass.charAt(i) == classToBeMatched.charAt(i); i++) {
             if (patternClass.charAt(i) == PACKAGE_SEPARATOR) {
                 result = i;
             }

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -185,7 +185,7 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
 
     private static int getLineStart(String line, final int tabWidth) {
         int lineStart = 0;
-        for (int index = 0; index < line.length(); ++index) {
+        for (int index = 0; index < line.length(); index++) {
             if (!Character.isWhitespace(line.charAt(index))) {
                 lineStart = CommonUtil.lengthExpandedTabs(line, index, tabWidth);
                 break;

